### PR TITLE
[#163190244] Adds accepted ToS version to user profile

### PR DESCRIPTION
**Pull request by chiadav**
[#163190244] L'accettazione dei Terms of Service va salvata nel profilo dell'utente

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#163190244](https://www.pivotaltracker.com/story/show/163190244): L'accettazione dei Terms of Service va salvata nel profilo dell'utente

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by fpersico** [comment]
@cloudify
There is another story to add also the accepted tos version. Can't we have just an optional accepted_tos_version field that if undefined is interpreted as no tos accepted?

File: api/definitions.yaml [hunk 1]
```diff
@@ -336,6 +339,11 @@ IsInboxEnabled:
   description: |-
     True if the recipient of a message wants to store its content for
     later retrieval.
+IsTosAccepted:
+  type: boolean
```
Comment (cloudify) [inline api/definitions.yaml]: we need to store the version (a number) of the last ToS accepted, the app will check this number against the latest ToS and if there's a newer ToS will ask the user to accept again

File: api/definitions.yaml [hunk 2]
```diff
@@ -322,6 +322,8 @@ ExtendedProfile:
       $ref: "#/PreferredLanguages"
     is_inbox_enabled:
       $ref: "#/IsInboxEnabled"
+    tos_version:
```
Comment (cloudify) [inline api/definitions.yaml]: perhaps `accepted_tos_version` is better, to make it clear that the user has accepted it

File: api/definitions.yaml [hunk 3]
```diff
@@ -336,6 +338,11 @@ IsInboxEnabled:
   description: |-
     True if the recipient of a message wants to store its content for
     later retrieval.
+TosVersion:
+  type: number
+  default: 0
```
Comment (cloudify) [inline api/definitions.yaml]: don't put a default, a missing value means the user has not accepted any Tos yet

File: api/definitions.yaml [hunk 4]
```diff
@@ -336,6 +338,11 @@ IsInboxEnabled:
...
+  description: |-
+    Version of terms of service.
```
Comment (cloudify) [inline api/definitions.yaml]:
```suggestion
    Version of latest terms of service accepted by the user.
```

File: api/definitions.yaml [hunk 5]
```diff
@@ -336,6 +338,11 @@ IsInboxEnabled:
   description: |-
     True if the recipient of a message wants to store its content for
     later retrieval.
+TosVersion:
+  type: number
```
Comment (cloudify) [inline api/definitions.yaml]:
```suggestion
  type: number
  minimum: 1
```

Comment (cloudify) [reply api/definitions.yaml]: set a minimum to make it a `NonNegativeNumber`

**Comment by cloudify** [comment]
@chiadav I've changed the title of this PR to make it more descriptive

File: lib/models/profile.ts [hunk 6]
```diff
@@ -3,6 +3,7 @@ import * as t from "io-ts";
 import { readonlySetType, tag } from "italia-ts-commons/lib/types";
 
 import * as DocumentDb from "documentdb";
+import { AcceptedTosVersion } from "../api/definitions/AcceptedTosVersion";
```
Comment (cloudify) [inline lib/models/profile.ts]: move this import below, along with the other imports from `"../api/definitions`

**Comment by cloudify** [comment]
LGTM

**Pull request merged**